libc/string_cpy.c

```diff
@@ -901,18 +943,43 @@ TEST(string_stpncpy, one_byte)
 }
 
 
+TEST(string_stpncpy, various_sizes)
```

Comment:
Is it possible to pass a size bigger than the source, or some other edge values there? If yes please add it